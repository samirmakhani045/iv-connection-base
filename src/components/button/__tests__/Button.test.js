import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Button from '../Button';

const defaultProps = {
  label: 'button',
  icon: 'Plus',
};

const Test = (props) => <Button {...defaultProps} {...props} />;

describe('<Button />', () => {
  it('should render correctly', () => {
    const { container } = render(<Test />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a button with variant: dark', () => {
    const { container } = render(<Test variant="dark" />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a button with variant: light', () => {
    const { container } = render(<Test variant="light" />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a small button', () => {
    const { container } = render(<Test small />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a disabled button', () => {
    const { container } = render(<Test disabled />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a disabled button with variant: dark', () => {
    const { container } = render(<Test disabled variant="dark" />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a disabled button with variant: light', () => {
    const { container } = render(<Test disabled variant="light" />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a disabled button with variant: ghost', () => {
    const { container } = render(<Test variant="ghost" />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render button with the icon on the right', () => {
    const { container } = render(<Test iconAlignRight />);

    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render button without label and icon/icon that does not exist', () => {
    render(<Test label={null} icon="ThisIconDoesNotExist" />);

    expect(screen.queryByTestId('button')).not.toBeInTheDocument();
  });

  it('should render icon if it does exist', () => {
    render(<Test icon="Add" />);

    expect(screen.queryByTestId('buttonIcon')).toBeInTheDocument();
  });

  it('should fire onClick if button has callback', () => {
    const onClick = jest.fn();
    render(<Test onClick={onClick} />);

    const button = screen.queryByTestId('button');

    userEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should not fire onClick if button has no callback', () => {
    const onClick = jest.fn();
    render(<Test />);

    const button = screen.queryByTestId('button');

    userEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  it('should not fire onClick if button is disabled', () => {
    const onClick = jest.fn();
    render(<Test disabled onClick={onClick} />);

    const button = screen.queryByTestId('button');

    userEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(0);
  });
});
