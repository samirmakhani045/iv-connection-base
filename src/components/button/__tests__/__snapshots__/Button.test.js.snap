// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button with variant: dark 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 iIsqkL"
  data-testid="button"
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render a button with variant: light 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 jPrzzZ"
  data-testid="button"
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render a disabled button 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 sokVo"
  data-testid="button"
  disabled=""
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render a disabled button with variant: dark 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 iLIhvI"
  data-testid="button"
  disabled=""
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render a disabled button with variant: ghost 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 cFFbDy"
  data-testid="button"
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render a disabled button with variant: light 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 dBWqvB"
  data-testid="button"
  disabled=""
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render a small button 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 eZirey"
  data-testid="button"
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render button with the icon on the right 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 dQPQGE"
  data-testid="button"
  type="button"
>
  <span>
    button
  </span>
</button>
`;

exports[`<Button /> should render correctly 1`] = `
<button
  class="Buttonstyles__StyledButton-sc-1uuuyg7-0 bhfzyd"
  data-testid="button"
  type="button"
>
  <span>
    button
  </span>
</button>
`;
